src/Microsoft.AspNetCore.Mvc.Razor/ViewComponentTagHelpers/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.Razor.Host;
+using Microsoft.AspNetCore.Mvc.ViewComponents;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.ViewComponentTagHelpers
+{
+    /// <summary>
+    /// Provides methods to create tag helper representations of view components.
+    /// </summary>
+    public class ViewComponentTagHelperDescriptorFactory
+    {
+        private readonly IViewComponentDescriptorProvider _descriptorProvider;
+
+        /// <summary>
+        /// Creates a new ViewComponentTagHelperDescriptorFactory than creates tag helper descriptors for
+        /// view components in the given descriptorProvider.
+        /// </summary>
+        /// <param name="descriptorProvider">The provider of view component descriptors.</param>
+        public ViewComponentTagHelperDescriptorFactory(IViewComponentDescriptorProvider descriptorProvider)
+        {
+            _descriptorProvider = descriptorProvider;
+        }
+
+        /// <summary>
+        /// Creates <see cref="TagHelperDescriptor"/> representations of view components in a given assembly.
```

Comment:
`in an <see href="Assembly"/> represented by the given <paramref name="assemblyName"/>`.